__init__.py

```diff
@@ -43,28 +46,51 @@ def __init__(self):
 
         self.nato_dict = None
 
+        self.mycroft_ready = False
+        self.pair_dialog_lock = Lock()
+        self.pairing_dialog = 'pairing.paired'
+
     def initialize(self):
-        # TODO:18.02 - use decorator
-        intent = IntentBuilder("PairingIntent") \
-            .require("PairingKeyword").require("DeviceKeyword").build()
-        self.register_intent(intent, self.handle_pairing)
         self.add_event("mycroft.not.paired", self.not_paired)
         self.nato_dict = self.translate_namedvalues('codes')
 
+        # If the device isn't paired catch mycroft.ready to report
+        # that the device is ready for use.
+        # This assumes that the pairing skill is loaded as a priority skill
+        # before the rest of the skills are loaded.
+        if not is_paired():
+            self.add_event("mycroft.ready", self.handle_mycroft_ready)
+
+        platform = self.config_core['enclosure'].get('platform', 'unknown')
+        if platform in PLATFORMS_WITH_BUTTON:
+            self.paired_dialog = 'pairing.paired'
+        else:
+            self.paired_dialog = 'pairing.paired.no.button'
+
+    def handle_mycroft_ready(self, message):
+        """ Catch info that skills are loaded and ready. """
+        with self.pair_dialog_lock:
+            if is_paired():
+                self.speak_dialog(self.paired_dialog)
+            else:
+                self.mycroft_ready = True
+
     def not_paired(self, message):
```

Comment:
if a variable is not being used, consider replacing it with an underscore.  In python, this means that a variable is expected there but there is no use for it.  This could be done throughout this class.